From 03f116d42ad75715e2abfb7aa16ab657389c7d21 Mon Sep 17 00:00:00 2001
From: Florian Scandella <flo@chilicode.com>
Date: Fri, 14 May 2010 16:31:14 +0200
Subject: [PATCH 3/3] use png_uint_32 for icc_profile_size.

casting the address of a 64 bit value to a pointer to a 32 bit value leaves
the upper or lower part of the 64 bit value undefined (depending on the byte order
i guess).
---
 gdk-pixbuf/io-png.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/gdk-pixbuf/io-png.c b/gdk-pixbuf/io-png.c
index 6cd0719..647b0ae 100644
--- a/gdk-pixbuf/io-png.c
+++ b/gdk-pixbuf/io-png.c
@@ -260,7 +260,7 @@ gdk_pixbuf__png_image_load (FILE *f, GError **error)
         gchar *icc_profile_base64;
         const gchar *icc_profile_title;
         const gchar *icc_profile;
-        gulong icc_profile_size;
+        png_uint_32 icc_profile_size;
         guint32 retval;
         gint compression_type;
 
@@ -341,9 +341,9 @@ gdk_pixbuf__png_image_load (FILE *f, GError **error)
         /* Extract embedded ICC profile */
         retval = png_get_iCCP (png_ptr, info_ptr,
                                (png_charpp) &icc_profile_title, &compression_type,
-                               (png_charpp) &icc_profile, (png_uint_32*) &icc_profile_size);
+                               (png_charpp) &icc_profile, &icc_profile_size);
         if (retval != 0) {
-                icc_profile_base64 = g_base64_encode ((const guchar *) icc_profile, icc_profile_size);
+                icc_profile_base64 = g_base64_encode ((const guchar *) icc_profile, (gsize) icc_profile_size);
                 gdk_pixbuf_set_option (pixbuf, "icc-profile", icc_profile_base64);
                 g_free (icc_profile_base64);
         }
-- 
1.7.1

